### ISSUE TITLE
[💡 Feature]: Export more of vitest to allow Component Testing users to mock external dependecies

### ISSUE THREAD

**Issue by nextlevelbeard** [webdriverio/webdriverio#9814]
### Is your feature request related to a problem?

There is no clear and documented way to mock external dependecies.
In `vitest` we can use `vi.mock` to mock specific properties or functions exported by a module.
But `@wdio/browser-runner` only seems to export `fn` and `spyOn` from ´@vitest/spy´.

### Describe the solution you'd like.

Allows users to import [vi.mock](https://vitest.dev/api/vi.html#vi-mock) and other `vitest` utilities.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by christian-bromann** [comment]
@nextlevelbeard great idea, can you provide a clear description of your use case? Which dependencies do you want to mock?

**Comment by nextlevelbeard** [comment]
A good use case would be testing a component that uses an external dependency to authenticate the user.
Spying here isn't enough.

As an example, [here](https://stackoverflow.com/questions/72732768) is someone asking how to mock and test something like this with `vitest`.

And here is how I imagine using it.

```js
import { expect } from '@wdio/globals'
import { render, screen } from '@testing-library/react'
import { fn, spyOn, mock } from '@wdio/browser-runner'

import * as ExternalAuth from 'someAuthProvider'

mock('someAuthProvider')

import MyComponent from "./my-component";

describe('MyComponent', () => {
  it('User email is displayed when MyComponent is rendered', async () => {

    ExternalAuth.doStuff = vi.fn().mockReturnValue({
      user: { 
        email : "user@example.com"
      },
    });
      
    // MyComponent uses method doStuff of module someAuthProvider
    render(<MyComponent />);

    const email = $(screen.getByTestId('email'))
    expect(email).toBeDisplayed()
  });
})
```

Instead of just importing specifc method we should simply export the entirety of [`vi`](https://vitest.dev/api/vi.html) that is possible and makes sense in the `@wdio/browser-runner`. There are other functions that might also come in handy.

Currently, if you try to just import `import { vi } from 'vitest'` to obtain `vi.mock` the test will have a `Uncaught TypeError: Cannot redefine property: Symbol($$jest-matchers-object)`

**Comment by christian-bromann** [comment]
> Spying here isn't enough.

Why is that? You should also be able to modify the behavior of a spy, right?

> And here is how I imagine using it.

This would pretty much work already:

```ts
    ExternalAuth.doStuff = fn().mockReturnValue({
      user: { 
        email : "user@example.com"
      },
    });
```

> There are other functions that might also come in handy.

Which one?

> Currently, if you try to jus `import { vi } from 'vitest'` to obtain `vi.mock` the test will have a `Uncaught TypeError: Cannot redefine property: Symbol($$jest-matchers-object)`

Note that Vitest runs in Node where it is feasible to mock modules in runtime. This is much more difficult when running things in the browser , though not impossible which is why I am want to know as much as possible about the use case.

**Comment by nextlevelbeard** [comment]
> This would pretty much work already

From my side I haven't been able to, here's some use cases and their errors.

---

Case 1
```ts
import { fn, spyOn } from '@wdio/browser-runner'

import * as ExternalAuth from 'someAuthProvider'

describe('MyComponent', () => {
  it('Test', async () => {
  
    ExternalAuth.doStuff = fn().mockImplementation(() => {({
      user: { 
        email : "user@example.com"
      }
    })})
   
  });
})
```
errors with `Cannot assign to read only property 'doStuff' of object '[object Module]'`

---

Case 2
```ts
import { fn, spyOn } from '@wdio/browser-runner'

import * as ExternalAuth from 'someAuthProvider'

describe('MyComponent', () => {
  it('Test', async () => {
  
    ExternalAuth.doStuff = fn().mockReturnValue({
      user: { 
        email : "user@example.com"
      }
    });
    
  });
})

```
errors with `Cannot assign to read only property 'doStuff' of object '[object Module]'`

---

Case 3:
```ts
import { fn, spyOn } from '@wdio/browser-runner'

import * as ExternalAuth from 'someAuthProvider'

describe('MyComponent', () => {
  it('Test', async () => {
    
    spyOn(ExternalAuth, 'doStuff').mockImplementation(() => {({
      user: { 
        email : "user@example.com"
      }
    })});

  });
})

```
errors with `TypeError: Cannot redefine property: doStuff`

**Comment by christian-bromann** [comment]
This will be implemented in #9821
